### PR TITLE
Feat/add balancer pools labels

### PR DESCRIPTION
**Pull request by Stefenon**
Brief comments on the purpose of your changes:

cc: @markusbkoch @mendesfabio @hedgehog-jacek 

Add Balancer V2 pools labels.

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/labels/balancer_v2/labels_balancer_v2_pools_arbitrum.sql [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        ON c.evt_tx_hash = cc.call_tx_hash
+    )
+
+    UNION ALL
+
+    SELECT pool_id, zip.tokens AS token_address,  zip.normalizedWeights/pow(10, 18) AS normalized_weight, symbol, pool_type
+    FROM (
+        SELECT c.poolId AS pool_id, explode(arrays_zip(cc.tokens, cc.normalizedWeights)) AS zip, cc.symbol, 'WP' AS pool_type
+        FROM {{ source('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }} c
+        INNER JOIN {{ source('balancer_v2_arbitrum', 'WeightedPoolV2Factory_call_create') }} cc
```
Comment (Hosuke) [inline models/labels/balancer_v2/labels_balancer_v2_pools_arbitrum.sql:29]: Please follow the source syntax and modify source tables used in the other files.

File: models/labels/balancer_v2/labels_balancer_v2_pools_ethereum.sql [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    SELECT c.poolId AS pool_id, explode(cc.tokens) AS token_address, NULL AS normalized_weight, cc.symbol, 'SP' AS pool_type
+    FROM balancer_v2_ethereum.Vault_evt_PoolRegistered c
+    INNER JOIN balancer_v2_ethereum.StablePoolFactory_call_create cc
+    ON c.evt_tx_hash = cc.call_tx_hash
+
+    UNION ALL
+
+    SELECT c.poolId AS pool_id, explode(cc.tokens) AS token_address, NULL AS normalized_weight, cc.symbol, 'SP' AS pool_type
+    FROM balancer_v2_ethereum.Vault_evt_PoolRegistered c
+    INNER JOIN balancer_v2_ethereum.MetastablePoolFactory_call_create cc
```
Comment (Hosuke) [inline models/labels/balancer_v2/labels_balancer_v2_pools_ethereum.sql:57]:
```suggestion
    FROM {{ source('balancer_v2_ethereum', 'Vault_evt_PoolRegistered') }} c
    INNER JOIN {{ source('balancer_v2_ethereum', 'MetastablePoolFactory_call_create') }} cc
```

File: models/labels/balancer_v2/labels_balancer_v2_sources.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+version: 2
+
+sources:
+  - name: balancer_v2_arbitrum
+    tables:
+      - name: InvestmentPoolFactory_call_create
+      # TODO Add and define source tables here
+      #  columns:
```
Comment (Hosuke) [inline models/labels/balancer_v2/labels_balancer_v2_sources.yml:8]: After source tables added, we can define the undefined(not defined before in the spellbook project) source tables in the `sources.yml` file, like we did for other balancer spells. https://github.com/duneanalytics/spellbook/blob/main/models/balancer/ethereum/balancer_ethereum_sources.yml

**Review by Hosuke — CHANGES_REQUESTED** [review]
Hi @Stefenon, please rewrite the source tables used in dbt source syntax, and define the undefined source in the sources.yml file.

File: models/labels/balancer_v2/labels_balancer_v2_pools_arbitrum.sql (hunk 1 above)
Comment (Stefenon) [reply models/labels/balancer_v2/labels_balancer_v2_pools_arbitrum.sql:29]: Thanks for noticing that @Hosuke, just commited these changes.

File: models/labels/balancer_v2/labels_balancer_v2_sources.yml (hunk 3 above)
Comment (Stefenon) [reply models/labels/balancer_v2/labels_balancer_v2_sources.yml]: I pushed a commit adding these source tables to the .yml file and another one updating the description of the sources that already existed on the balancer_v2 `sources.yml` file folders. @mendesfabio please review if all descriptions are correct.

**Comment by mendesfabio** [comment]
LGTM!

File: models/labels/balancer_v2/labels_balancer_v2_pools_ethereum.sql [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    )
+
+    UNION ALL
+
+    SELECT pool_id, zip.tokens AS token_address,  zip.weights/pow(10, 18) AS normalized_weight, symbol, pool_type
+    FROM (
+        SELECT c.poolId AS pool_id, explode(arrays_zip(cc.tokens, cc.weights)) AS zip, cc.symbol, 'WP2T' AS pool_type
+        FROM {{ source('balancer_v2_ethereum', 'Vault_evt_PoolRegistered') }} c
+        INNER JOIN {{ source('balancer_v2_ethereum', 'WeightedPool2TokensFactory_call_create') }} cc
+        ON c.evt_tx_hash = cc.call_tx_hash
```
Comment (Hosuke) [inline models/labels/balancer_v2/labels_balancer_v2_pools_ethereum.sql:33]:
```suggestion
        SELECT c.poolId AS pool_id, explode(arrays_zip(cc.tokens, cc.weights)) AS zip, cc.symbol, 'WP2T' AS pool_type
        FROM {{ source('balancer_v2_ethereum', 'Vault_evt_PoolRegistered') }} c
        INNER JOIN {{ source('balancer_v2_ethereum', 'WeightedPool2TokensFactory_call_create') }} cc
        ON c.evt_tx_hash = cc.call_tx_hash
        {% if is_incremental() %}
        WHERE c.evt_block_time >= date_trunc("day", now() - interval '1 week')
          AND cc.call_block_time >= date_trunc("day", now() - interval '1 week')
        {% endif %}
```

**Review by Hosuke — CHANGES_REQUESTED** [review]
We can make the pools_label as incremental views by adding incremental filters.

**Comment by Hosuke** [comment]
Hi @Stefenon , In my last commits I try to modify `labels_balancer_v2_pools_arbitrum` to incremental view, with `address` as unique key.
Please have a review as I may miss or break something.

**Comment by Stefenon** [comment]
> 

Hey @Hosuke, the changes look good to me. I changed the other queries similarly, and I believe they should be good to go now.

**Comment by Stefenon** [comment]
> Hi @Stefenon , In my last commits I try to modify `labels_balancer_v2_pools_arbitrum` to incremental view, with `address` as unique key. Please have a review as I may miss or break something.

Hey @Hosuke, just changed the other queries following this suggestion. However, the `pools_ethereum` label didn't pass the unique column test. I checked the result and it turns out there are two pools which apparently were created in the same transaction (https://etherscan.io/address/0x76fcf0e8c7ff37a47a799fa2cd4c13cde0d981c9 and https://etherscan.io/address/0xd1ec5e215e8148d76f4460e4097fd3d5ae0a3558). I will investigate this issue further so that the queries can be fixed.

**Comment by Stefenon** [comment]
> > Hi @Stefenon , In my last commits I try to modify `labels_balancer_v2_pools_arbitrum` to incremental view, with `address` as unique key. Please have a review as I may miss or break something.
> 
> Hey @Hosuke, just changed the other queries following this suggestion. However, the `pools_ethereum` label didn't pass the unique column test. I checked the result and it turns out there are two pools which apparently were created in the same transaction (https://etherscan.io/address/0x76fcf0e8c7ff37a47a799fa2cd4c13cde0d981c9 and https://etherscan.io/address/0xd1ec5e215e8148d76f4460e4097fd3d5ae0a3558). I will investigate this issue further so that the queries can be fixed.

I fixed the queries so that problems such as this one don't happen anymore. I believe the spells now are good to go.

**Review by Hosuke — APPROVED** [review]
This PR adds balancer_v2 pools labels.
I have reviewed the PR and here is the summary:

- Metadata & naming conventions look good ✅
- Config blocks match uniqueness tests in schema files ✅
- Incremental filters added when joining large tables ✅
- Inner join to large tables as possible ✅

[arbitrum pools sample](https://dune.com/queries/1922937)
[ethereum pools sample](https://dune.com/queries/1922940)
[polygon pools sample](https://dune.com/queries/1922941)
[optimism pools sample](https://dune.com/queries/1922942)
[balancer_v2_pools sample](https://dune.com/queries/1922930?d=1)

**Comment by soispoke** [comment]
lgtm, thanks!

**Pull request merged**